x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java

```diff
@@ -439,20 +439,20 @@ private void buildUser(SamlAttributes attributes, ActionListener<AuthenticationR
             return;
         }
 
-        final Map<String, Object> userMeta = new HashMap<>();
+        final Map<String, Object> userMetaBuilder = new HashMap<>();
         if (populateUserMetadata) {
             for (SamlAttributes.SamlAttribute a : attributes.attributes()) {
-                userMeta.put("saml(" + a.name + ")", a.values);
+                userMetaBuilder.put("saml(" + a.name + ")", a.values);
                 if (Strings.hasText(a.friendlyName)) {
-                    userMeta.put("saml_" + a.friendlyName, a.values);
+                    userMetaBuilder.put("saml_" + a.friendlyName, a.values);
                 }
             }
         }
         if (attributes.name() != null) {
-            userMeta.put(USER_METADATA_NAMEID_VALUE, attributes.name().value);
-            userMeta.put(USER_METADATA_NAMEID_FORMAT, attributes.name().format);
+            userMetaBuilder.put(USER_METADATA_NAMEID_VALUE, attributes.name().value);
+            userMetaBuilder.put(USER_METADATA_NAMEID_FORMAT, attributes.name().format);
         }
-
+        final Map<String, Object> userMeta = Map.copyOf(userMetaBuilder);
```

Comment:
`Map.copyOf` requires that the source map does not contain any `null` values, and we don't know that here.
I'm pretty sure it is possible for `USER_METADATA_NAMEID_FORMAT` to have null value.

Reply:
> I'm pretty sure it is possible for USER_METADATA_NAMEID_FORMAT to have null value.

You're right, I will do the check.